subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -549,17 +549,22 @@ struct ztest_unit_test *z_ztest_get_next_test(const char *suite, struct ztest_un
 #ifdef CONFIG_ZTEST_SHUFFLE
 static void z_ztest_shuffle(void *dest[], intptr_t start, size_t num_items, size_t element_size)
 {
+	void *tmp;
+
+	/* Initialize dest array */
 	for (size_t i = 0; i < num_items; ++i) {
-		int pos = sys_rand32_get() % num_items;
-		const int start_pos = pos;
```

Comment:
As a quick and dirty alternative, we could also just add `__maybe_unused` with a comment about asserts not being defined if we want to keep the `__ASSERT_NO_MSG(pos != start_pos);`

Reply:
That was my original fix, but I wanted to revert to my original implementation of randomizing an array which gives a better distribution.